interfaces/cython/cantera/test/test_thermo.py

```diff
@@ -1232,6 +1232,77 @@ def test_standalone(self):
         self.assertEqual(len(c), 2)
         self.assertEqual(c['C'], 1)
         self.assertEqual(c['H'], 4)
+        self.assertNear(s.molecular_weight, 16.043)
+
+    def test_molecular_weight_with_unstable_element(self):
+        s = ct.Species("CPo", {"C": 1, "Po": 1})
+        with pytest.raises(ct.CanteraError, match="has no stable isotopes"):
+            s.molecular_weight
+
+    def test_molecular_weight_with_electron(self):
+        yaml = """
+            name: Li+[elyt]
+            composition: {Li: 1, E: -1}
+        """
+        s = ct.Species.from_yaml(yaml)
+        assert np.isclose(s.molecular_weight, 6.939451420091127)
+
+    def test_custom_element_weight_is_set_on_species(self):
+        gas = ct.Solution("ideal-gas.yaml", "element-override")
+        # Check that the molecular weight stored in the phase definition is the same
+        # as the one on the element
+        self.assertNear(
+            gas["AR"].molecular_weights[0], gas.species("AR").molecular_weight
+        )
+        # Check that the custom value is actually used
+        self.assertNear(gas.species("AR").molecular_weight, 36.0)
+
+    def test_species_can_be_added_to_phase(self):
+        s = ct.Species.from_dict({
+            "name": "AR2",
+            "composition": {"Ar": 2},
+            "thermo": {"model": "constant-cp", "h0": 100}
+        })
+        # Access the molecular weight to make sure it's been computed by the Species
+        self.assertNear(s.molecular_weight, 39.95 * 2)
+        # This should not cause an error because the existing Ar definition is the
+        # default molecular weight
+        self.gas.add_species(s)
+        self.assertNear(
+            self.gas["AR2"].molecular_weights[0],
+            self.gas.species("AR2").molecular_weight
+        )
+
+    def test_species_cannot_be_added_to_phase_custom_element(self):
+        s = ct.Species.from_dict({
+            "name": "AR2",
+            "composition": {"Ar": 2},
+            "thermo": {"model": "constant-cp", "h0": 100}
+        })
+        # Access the molecular weight to make sure it's been computed by the Species
+        self.assertNear(s.molecular_weight, 39.95 * 2)
+        gas = ct.Solution("ideal-gas.yaml", "element-override")
+        # The error here is because the weight of the Argon element has been changed in
+        # the phase definition, but the molecular weight of the species has already been
+        # computed
+        with pytest.raises(ct.CanteraError, match="Molecular weight.*changed"):
+            gas.add_species(s)
```

Comment:
The point is that the species definition _does_ include information about molecular weight now, since the `molecular_weight` method was called. Maybe I'm not understanding the question?

Reply:
I was referring to the YAML definition - the user doesn't know how things are implemented under the hood.

Reply:
Sorry, I'm still not understanding what you mean. The YAML definition of a species implicitly contains information about the molecular weight, since it includes the elemental composition. This is testing for the case where a species has been created in Python, then added to a phase.

Reply:
I am mostly thinking about user intent here - at the end of the day, we're just discussing whether this should raise a warning or throw an error.

Reply:
:+1: Then let's move the discussion to that comment below 😄